### PR TITLE
feat(gstd): optimize calls to const syscalls

### LINKED ISSUE #1467: Optimize calls to const syscalls
`gr_origin`
`gr_program_id`
`gr_size`
`gr_source`
`gr_value`

but caching them should occur per-message

### PR DESCRIPTION
**Pull request by StackOverflowExcept1on**
Resolves #1467

This is proof of concept for caching const syscalls. We probably need to remove the imports: `gr_is_getter_call`, `gr_set_getter_call` from the wasm binary instead of keeping dangling imports like we do now.

@gear-tech/dev

### CODE REVIEW

File: core-backend/sandbox/src/env.rs [hunk 1]
```diff
@@ -260,6 +260,8 @@ where
 
         builder.add_func(Alloc, wrap_common_func!(FuncsHandler::alloc, (1) -> (1)));
         builder.add_func(Free, wrap_common_func!(FuncsHandler::free, (1) -> (1)));
+        builder.add_func(IsGetterCalled, wrap_common_func!(FuncsHandler::forbidden, () -> ()));
```
Comment (grishasobol) [inline core-backend/sandbox/src/env.rs:263]: Why do we need this sys-calls support on back-end if they were removing on instrumentation stage ?

**Review by grishasobol — CHANGES_REQUESTED** [review]
Blocked, until it's not discussed and synced with other possible implementations.

File: core-backend/sandbox/src/env.rs (hunk 1 above)
Comment (StackOverflowExcept1on) [reply core-backend/sandbox/src/env.rs:263]: I'm working on that. We probably need to add it to the list of forbidden functions. This causes the error right now: "Not all existing sys-calls were added to the module's env"

**Comment by gshep** [comment]
To tell the truth such kind of caching could be done explicitly on the program side

**Pull request closed without merging**